lrpar/src/lib/ctbuilder.rs

```diff
@@ -190,6 +190,20 @@ impl ToTokens for Visibility {
     }
 }
 
+impl Visibility {
+    fn to_variant_tokens(&self, tokens: &mut TokenStream) {
+        tokens.extend(match self {
+                Visibility::Private => quote!(::lrpar::Visibility::Private),
+                Visibility::Public => quote!(::lrpar::Visibility::Public),
+                Visibility::PublicSuper => quote!(::lrpar::Visibility::PublicSuper),
+                Visibility::PublicSelf => quote!(::lrpar::Visibility::PublicSelf),
+                Visibility::PublicCrate => quote!(::lrpar::Visibility::PublicCrate),
+                Visibility::PublicIn(data) => quote!(::lrpar::Visibility::PublicIn(#data)),
```

Comment:
This needs the `QuoteToString` type from `lrlex::ctbuilder`.
But even then that won't work because `to_string()` cannot be used in a const context.

Reply:
So really the only way I know how to fix that is to either include a lifetime `Visibility<'a>` so we can say `Visibility::PublicIn(&'a str)`  or change it to `Visibility::PublicIn(&'static str)`.

Maybe It is best to just go back to `VISIBILITY: &'static str = stringify(#visibility)`.

Reply:
Hmm, the `'a` lifetime here could make the `enum` a pain to use in some situations for external users?

Reply:
I agree that `'a` and `'static` feel like they're pushing an internal detail (how quote works) to external users, and that if we can avoid burdening external users it would be better.

Reply:
Just to clarify it isn't exactly a quote implementation detail.
for instance we can reproduce the issue entirely without quote by trying to construct it in a `const` fashion

```
  const V: Visibility = Visibility::PublicIn("crate".to_string());
```

This is basically running into the same error as the following, because string isn't `const` due to heap allocations.
```
const X: String = "y".to_string();
```

It occurs to me that the following is fine:
```
quote!{
    fn visibility() -> Visibility {
       ::lrpar::Visibility::PublicIn(#data.to_string())
    }
}
```

So perhaps an alternative to `stringify!` is to make this case a function.

Reply:
> So perhaps an alternative to stringify! is to make this case a function.

If so, that sounds good!

Reply:
Should be fixed in [7e9a17b](https://github.com/softdevteam/grmtools/pull/504/commits/7e9a17b901930fc96bff9df295f05fcceda22522)